.github/workflows/ci.yml

```diff
@@ -0,0 +1,173 @@
+---
+name: CI
+on:
+  pull_request:
+jobs:
+  api-test:
+    runs-on: ubuntu-latest
+    permissions:
+      packages: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Log in to registry
+        run: |
+          echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+
+      - name: Pre-pull image to warm build cache
+        run: |
+          docker pull ghcr.io/${{ github.repository_owner }}/awx_devel:${{ github.base_ref }}
+
+      - name: Build image
+        run: |
+          DEV_DOCKER_TAG_BASE=ghcr.io/${{ github.repository_owner }} COMPOSE_TAG=${{ github.base_ref }} make docker-compose-build
```

Comment:
These 3 steps are replicated 9-ish times, and I see in the output that many of the checks have this as a bottleneck for timing. For good practice adoption, I would like to see if I could refactor that into a composite action.

https://github.blog/changelog/2020-08-07-github-actions-composite-run-steps/

Things totally work now and I'm fine to move ahead with this, but I'd like to leave the door open for someone (myself included) to thinker with this.